types/types.cc

```diff
@@ -349,18 +349,25 @@ static uint32_t serialize(sstring_view input, int64_t days) {
     return static_cast<uint32_t>(days);
 }
 uint32_t simple_date_type_impl::from_sstring(sstring_view s) {
-    char* end;
-    auto v = std::strtoll(s.begin(), &end, 10);
-    if (end == s.begin() + s.size()) {
-        return v;
-    }
-    static const boost::regex date_re("^(-?\\d+)-(\\d+)-(\\d+)");
-    boost::match_results<sstring_view::const_iterator> dsm;
-    if (!boost::regex_match(s.begin(), s.end(), dsm, date_re)) {
-        throw marshal_exception(format("Unable to coerce '{}' to a formatted date (long)", s));
-    }
-    auto t = get_simple_date_time(dsm);
-    return serialize(s, date::local_days(t).time_since_epoch().count());
+    try {
```

Comment:
I am guessing you added this try because C++'s std::strtoll() throws instad of returning a number like the old C function? If this is the case, please wrap only that call, not all the code in this function, it will make the code clearer (and you won't need to reindent it).

Reply:
This is surprising, I seem to remember the C++ function does throw. Please add a test for this case - a number like 50000000000000000000000 which doesn't even fit in 64 bit, and see how this case behaves.

Reply:
`strtoll()` - it comes from C and does not throw. It is defined in header: `<cstdlib>`.
`stoll()` - it comes from C++, internally it calls `strtoll()` and throws if the exit code is invalid. It is defined in header: `<string>`.

C++17 introduced `std::from_chars`. It does not throw exceptions and in some cases it may be faster (especially floats conversion). I am not aware how it compares against `std::strtoll`.